README.md

```diff
@@ -16,184 +16,76 @@
-
-# Kubernetes Instructions
------------------------
-
-You can run Polaris as a mini-deployment locally. This will create two pods that bind themselves to port `8181`:
-
-```
-./setup.sh
-```
-
-You can check the pod and deployment status like so:
-
-```
-kubectl get pods
-kubectl get deployment
-```
-
-If things aren't working as expected you can troubleshoot like so:
-
-```
-kubectl describe deployment polaris-deployment
-```
-
-## Creating a Catalog manually
-
-Before connecting with Spark, you'll need to create a catalog. To create a catalog, generate a token for the root
-principal:
-
+ 
+Polaris is built using Gradle with Java 21+ and Docker 27+.
+- `./gradlew build` - To build and run tests. Make sure Docker is running, as the integration tests depend on it.
+- `./gradlew assemble` - To skip tests.
+- `./gradlew test` - To run unit tests and integration tests.
+- `./gradlew runApp` - To run the Polaris server locally on localhost:8181. 
+  - The server starts with the in-memory mode, and it prints the auto-generated credentials to STDOUT in a message like this `realm: default-realm root principal credentials: <id>:<secret>`
+  - These credentials can be used as "Client ID" and "Client Secret" in OAuth2 requests (e.g. the `curl` command below).
+- `./regtests/run.sh` - To run regression tests or end-to-end tests in another terminal.
+
+Running in Docker
+- `docker build -t localhost:5001/polaris:latest .` - To build the image.
+- `docker run -p 8181:8181 localhost:5001/polaris:latest` - To run the image in standalone mode.
+- `docker compose up --build --exit-code-from regtest` - To run regression tests in a Docker environment.
+
+Running in Kubernetes
+- `./setup.sh` - To run Polaris as a mini-deployment locally. This will create two pods that bind themselves to port `8181`.
+- `kubectl get pods` - To check the status of the pods.
+- `kubectl get deployment` - To check the status of the deployment.
+- `kubectl describe deployment polaris-deployment` - To troubleshoot if things aren't working as expected.
+
+Building docs
+- Docs are generated using [Redocly](https://redocly.com/docs/cli/installation). To regenerate them, run the following
+commands from the project root directory.
 ```bash
-curl -i -X POST \
-  http://localhost:8181/api/catalog/v1/oauth/tokens \
-  -d 'grant_type=client_credentials&client_id=<principalClientId>&client_secret=<mainSecret>&scope=PRINCIPAL_ROLE:ALL'
-```
-
-The response output will contain an access token:
-
-```json
-{
-  "access_token": "ver:1-hint:1036-ETMsDgAAAY/GPANareallyverylongstringthatissecret",
-  "token_type": "bearer",
-  "expires_in": 3600
-}
+docker run -p 8080:80 -v ${PWD}:/spec docker.io/redocly/cli join spec/docs.yaml spec/polaris-management-service.yml spec/rest-catalog-open-api.yaml -o spec/index.yaml --prefix-components-with-info-prop title
+docker run -p 8080:80 -v ${PWD}:/spec docker.io/redocly/cli build-docs spec/index.yaml --output=docs/index.html --config=spec/redocly.yaml
```

Comment:
LGTM! This is a much tidier way of presenting this info -- agree with you that the earlier links are no longer relevant now that everything is on a single GitHub Page. 🚢